# Project Title: YouTube Keyword Prediction for Trending Videos


### Loading Nigeria's Current Trending YouTube Videos:

###### Importing packages and reading data

In [1]:
# importing important libararies
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns


sns.set()

In [2]:
#importing the NigeriaYoutube data 
Ng_data= pd.read_csv('NG_youtube_data.csv')

In [3]:
Ng_data.head(4)

title      videoId  \
0  FAMILY MATTERS | Brodashaggi | Mr Ibu | Kiriku...  Zv1i-Cdn4Ik   
1  Best Friends in the World: Senior Year | Episo...  BmCkK7W7j0Q   
2  ROMOLA ALAKARA - Latest 2022 Yoruba Movie Star...  seZb9P7H6MU   
3                                       Money & Love  pbATe_uZFTE   

                  channelId           channelTitle  categoryId  \
0  UCG6orNVuXIICv9_ifH6msIA  official BRODA SHAGGI          23   
1  UCCl71WmqkG8BjVuySOSCh_Q       Neptune3 Studios           1   
2  UCK2-495cMvdhaMqXcLlgmcg             YORUBAPLAY          24   
3  UC25DGtgYk5TwdksvtpeWYvA         Wizkid - Topic          10   

  VideoCategoryDescription                                        Description  \
0                   Comedy  Please SUBSCRIBE to Broda Shaggi Official YouT...   
1         Film & Animation  EP 13 "The Protest" - Olive leads a protest ag...   
2            Entertainment  Find out where the story leads in this latest ...   
3                    Music  Provided to YouTube by Starboy/RCA Records\n\n...   

   viewCount  likeCount  dislikeCount  commentCount  favoriteCount  \
0     654033      30422             0          1474              0   
1     299169      21292             0          2012              0   
2     553886       5252             0           586              0   
3     197492       8561             0             0              0   

  publishedAt   duration  
0  22/10/2022   PT10M18S  
1  28/10/2022  PT1H5M22S  
2  21/10/2022  PT1H9M27S  
3  27/10/2022    PT3M12S

### Loading United State's Current Trending YouTubeVideos: 
###### Importing packages and reading data

In [4]:
# importing the us data 
US_data = pd.read_csv('us_Youtube_data.csv')

In [5]:
US_data.head()

channelId           channelTitle      videoId  \
0  UCU1_l0ZJyTK_7HZZ3Ruw8Dg                   MAPS  pTnk3ziVVRM   
1  UCLuO2lUqHrPIIpx0hFenV2g         Tink Tink Club  cuJjSeHZIrg   
2  UCihqrkaOgVMfLNo2W1hSliA           Podcast Bunk  IuyuZfWtGgg   
3  UCgbWWPn3VYYzxjffZbfj9GQ        Alan Springwind  cng_ZhQf8iY   
4  UCFmLi6X1mojkFZOFngNR9tQ  Drug Education Agency  OpQIQEx7J5A   

                publishedAt  \
0  2014-01-10T01:24:57.000Z   
1  2015-06-18T16:56:04.000Z   
2  2016-05-01T05:33:13.000Z   
3  2016-01-25T04:48:22.000Z   
4  2014-08-15T10:53:58.000Z   

                                          videoTitle  \
0  Psychedelic Horizons Beyond Psychotherapy Work...   
1                     Episode 35 - Dr. James Fadiman   
2  #325 Microdosing from The Adam and Dr Drew Sho...   
3                Microdosing Away The Same Old Blues   
4  Erschossener Kiffer / Drogen in Mikro-Dosierun...   

                                    videoDescription  videoCategoryId  \
0  Watch the full workshop at http://psychedelics...               29   
1  Dr. James Fadiman is the father of modern psyc...               22   
2  Adam and Dr. Drew are solo today and they open...               22   
3  Source: https://www.spreaker.com/user/springwi...               24   
4  Von erschossenen "Dealern", von demonstrierend...               24   

      videoCategoryLabel    duration  durationSec definition  caption  \
0  Nonprofits & Activism  PT1H19M40S         4780         hd    False   
1         People & Blogs  PT1H12M34S         4354         sd    False   
2         People & Blogs    PT50M40S         3040         sd    False   
3          Entertainment    PT32M21S         1941         hd    False   
4          Entertainment    PT26M32S         1592         hd    False   

   viewCount  likeCount  dislikeCount  commentCount  
0       1512        8.0           0.0           1.0  
1        881       14.0           0.0           3.0  
2         67        0.0           1.0           3.0  
3          4        0.0           0.0           0.0  
4       5465      121.0           6.0          25.0

### Sorting Needed Columns For Nigeria's Current Trending YouTube Video Dataset:
###### Here we are creating a dataframe that shows the trending Nigerian Dataset by Video Category Decription and View Counts in ascending order.

###### So we have 7 trending Nigerian YouTube Video Categories and they are as follows:
       Comedy, Entertainment, Film & Animation, Music, News & Politics, People & Blogs, and Sports.
    
###### The highest trending video under Comedy has 1,629,458 viewCounts
###### The highest trending video under <font color=red>Entertainment</font> has 34,543,612 viewCounts
###### The highest trending video under Film & Animation has 1,648,620 viewCounts
###### The highest trending video under Music has 12,173,797 viewCounts 
###### The highest trending video under News & Politics has 274,827 viewCounts
###### The highest trending video under People & Blogs has 574,391 viewCounts
###### The highest trending video under Sports has 7,154,769 viewCounts

###### Out of all the 7 trending Nigerian Youtube Video Categories, <font color=red>Entertainment</font> seems to have the most viewcount , therefore making it the highest trending youtube video category currently in Nigeria!


NigeriaYoutubeData = pd.read_csv('NG_youtube_data.csv')

In [6]:
# Sorting Values for the Nigeria Youtube data 
Ng_data.sort_values(by=['VideoCategoryDescription','viewCount','likeCount', 'title', 'Description'], ascending=(False,False,False,False,False)).head(7)

title      videoId  \
34  HIGHLIGHTS: Liverpool 1-0 Manchester City | Sa...  TE0e64xbtqQ   
13  UNBELIEVABLE Performance!  | Man Utd 2-0 Totte...  O0W4e7d4VvE   
37       Resumen de Real Madrid vs FC Barcelona (3-1)  swLI6zuXtp4   
22  A FIRST Goal For Casemiro  | Chelsea 1-1 Man U...  klG_itC5ROw   
5   Roma-Napoli 0-1 | Osimhen stuns Roma with a be...  Ga_i97g0uFY   
42  EVERTON 3-0 CRYSTAL PALACE | Premier League hi...  rND-d_m19Dg   
48                  Kizz daniel BUGA Wedding Entrance  oOzKbiWWl20   

                   channelId           channelTitle  categoryId  \
34  UC9LQwHZoucFT94I2h6JOcjw           Liverpool FC          17   
13  UC6yW44UGJJBvYTlfC7CRg2Q      Manchester United          17   
37  UCTv-XvfzLX3i4IGWAm4sbmA       LaLiga Santander          17   
22  UC6yW44UGJJBvYTlfC7CRg2Q      Manchester United          17   
5   UCBJeMCIeLQos7wacox4hmLQ                Serie A          17   
42  UCtK4QAczAN2mt2ow_jlGinQ  Everton Football Club          17   
48  UCQHJ0bJK5Z7m7EGbBoU-WDA      K.K.T Productions          22   

   VideoCategoryDescription  \
34                   Sports   
13                   Sports   
37                   Sports   
22                   Sports   
5                    Sports   
42                   Sports   
48           People & Blogs   

                                          Description  viewCount  likeCount  \
34  Watch all the key action from Liverpool's Prem...    7154769     113668   
13  Relive the utter dominance from the Reds under...    7068044     148312   
37  El Real Madrid se impone al FC Barcelona en El...    5826699     120124   
22  Take a look at the highlights from United's 1-...    3717183      88317   
5   The Nigerian striker smashes home the winner w...    1350957      21167   
42  Dominic Calvert-Lewin, Anthony Gordon and Dwig...     785081      12704   
48                                                NaN     574391      10640   

    dislikeCount  commentCount  favoriteCount publishedAt duration  
34             0          4920              0  16/10/2022  PT2M18S  
13             0          4342              0  19/10/2022  PT2M32S  
37             0          4345              0  16/10/2022   PT3M2S  
22             0          2550              0  22/10/2022  PT2M39S  
5              0          1043              0  23/10/2022  PT3M32S  
42             0           488              0  22/10/2022  PT2M21S  
48             0            46              0  16/10/2022    PT34S

# Sorting Needed Columns for United State's Current Trending Youtube Video Dataset:
###### Here we are creating a dataframe that shows the trending United State Dataset by Video Category Label and View Count in ascending order.

###### So we have 13 trending United State YouTube Video Categories and they are as follows:
       Comedy, Entertainment, Film & Animation, Music, News & Politics, People & Blogs, Sports, Education, Science & Technology, Howto & Style, Nonprofits & Activism, Gaming, and Travel & Events.
    
###### The highest trending video under Comedy has 515 viewCounts
###### The highest trending video under <font color=red>Entertainment</font> has 148,681 viewCounts
###### The highest trending video under Film & Animation has 4,040 viewCounts
###### The highest trending video under Music has 160 viewCounts 
###### The highest trending video under News & Politics has 108,974 viewCounts
###### The highest trending video under People & Blogs has 18,908 viewCounts
###### The highest trending video under Sports has 1,805 viewCounts
###### The highest trending video under  Education has 35,928 viewCounts
###### The highest trending video under Science & Technology has 30 viewCounts
###### The highest trending video under  Howto & Style has 294 viewCounts
###### The highest trending video under  Nonprofits & Activism has 1,512 viewCounts
###### The highest trending video under Gaming has 1,851 viewCounts
###### The highest trending video under Travel & Events has 843 viewCounts


###### Out of all the 13 trending United State Youtube Video Categories, <font color=red>Entertainment</font> again seems to have the most viewcount , therefore making it the highest trending youtube video category currently in the United States!



In [7]:
# sorting the United states Youtube data
US_data.sort_values(by=['videoCategoryLabel', 'videoTitle', 'videoDescription','viewCount', 'likeCount'], ascending=(True,False,False,False,False)).head(8)


channelId          channelTitle      videoId  \
61   UCZKcHstffFwoxnTs5uKmryw       kodie lallement  P1Zb1ixhesA   
64   UCfBmmf1OpYLL7JrD2ExMQLA      VIZAKIS MODALITY  tvtzuL45dbc   
44   UCWhd7Voh_8ZjyM2NryqpMbg  TheOriginsOfReligion  waKpmL2rfaQ   
10   UCXSF1F_RFRUVNXX4QmB6vmw         Your Mate Tom  cnFTo2zFUgw   
33   UCXSF1F_RFRUVNXX4QmB6vmw         Your Mate Tom  iG6nxxfx42w   
31   UCXSF1F_RFRUVNXX4QmB6vmw         Your Mate Tom  kbVwK9hqSKc   
114  UCH4ezCqRXHQhsETfl0dainA     FabFormIndustries  ODX4tzCWhhU   
105  UCZA4UoKQSLUzJ3g4DLZS4EA       wemakeideasflow  XqLro_GPPIc   

                  publishedAt  \
61   2014-01-18T12:10:49.000Z   
64   2016-04-11T05:16:48.000Z   
44   2015-10-12T23:19:55.000Z   
10   2015-12-05T16:26:46.000Z   
33   2015-11-10T09:20:00.000Z   
31   2015-11-13T14:35:06.000Z   
114  2015-08-19T18:25:20.000Z   
105  2013-04-11T17:04:03.000Z   

                                            videoTitle  \
61               con cret micro dosing creatine review   
64        Psychedelic Micro dosing and Weight Training   
44   Microdosing Psilocybe Testimony - The Natural ...   
10   Microdosing PSILOCYBIN Mushrooms Experiment ||...   
33   Microdosing PSILOCYBIN Mushrooms Experiment | ...   
31           Microdosing PSILOCYBIN Experiment | Day 2   
114                           Dosing Helix Micro Rebar   
105                           7615 Micro Dosing Pump-1   

                                      videoDescription  videoCategoryId  \
61   my opinion and review and concret micro dosing...               23   
64   Join me as I explore the world of micro dosing...               27   
44   Microdosing Psilocybe Testimony - The Natural ...               27   
10   I DO NOT CONDONE THE USE OF ANY DRUGS --------...               27   
33   Support me on Patreon: https://www.patreon.com...               27   
31   Microdosing PSILOCYBIN | Day 1 https://www.you...               27   
114  Coquitlam concrete goes into detail how they d...               27   
105                                                                  27   

    videoCategoryLabel  duration  durationSec definition  caption  viewCount  \
61              Comedy   PT3M10S          190         sd    False        515   
64           Education   PT2M53S          173         hd    False        263   
44           Education   PT4M55S          295         hd    False        611   
10           Education  PT12M32S          752         hd    False       8869   
33           Education   PT7M24S          444         hd    False      35928   
31           Education   PT7M32S          452         hd    False       5838   
114          Education      PT1M            1         hd    False        345   
105          Education     PT43S           43         sd    False        218   

     likeCount  dislikeCount  commentCount  
61         2.0           0.0           0.0  
64        12.0           0.0           0.0  
44         7.0           0.0           3.0  
10       298.0           1.0          80.0  
33       323.0          12.0         178.0  
31       113.0           0.0          26.0  
114        4.0           0.0           0.0  
105        0.0           0.0           0.0

In [8]:
# initial data info before cleaning the data
Ng_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50 entries, 0 to 49
Data columns (total 14 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   title                     50 non-null     object
 1   videoId                   50 non-null     object
 2   channelId                 50 non-null     object
 3   channelTitle              50 non-null     object
 4   categoryId                50 non-null     int64 
 5   VideoCategoryDescription  50 non-null     object
 6   Description               44 non-null     object
 7   viewCount                 50 non-null     int64 
 8   likeCount                 50 non-null     int64 
 9   dislikeCount              50 non-null     int64 
 10  commentCount              50 non-null     int64 
 11  favoriteCount             50 non-null     int64 
 12  publishedAt               50 non-null     object
 13  duration                  50 non-null     object
dtypes: int64(6), object(8)
memor

In [9]:
# initial data info before cleaning the data

US_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 115 entries, 0 to 114
Data columns (total 16 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   channelId           115 non-null    object 
 1   channelTitle        115 non-null    object 
 2   videoId             115 non-null    object 
 3   publishedAt         115 non-null    object 
 4   videoTitle          115 non-null    object 
 5   videoDescription    109 non-null    object 
 6   videoCategoryId     115 non-null    int64  
 7   videoCategoryLabel  115 non-null    object 
 8   duration            115 non-null    object 
 9   durationSec         115 non-null    int64  
 10  definition          115 non-null    object 
 11  caption             115 non-null    bool   
 12  viewCount           115 non-null    int64  
 13  likeCount           111 non-null    float64
 14  dislikeCount        111 non-null    float64
 15  commentCount        113 non-null    float64
dtypes: bool(

In [10]:
# subsetting the US data into 50 rows using .iloc
US_data = US_data.iloc[:50]
US_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50 entries, 0 to 49
Data columns (total 16 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   channelId           50 non-null     object 
 1   channelTitle        50 non-null     object 
 2   videoId             50 non-null     object 
 3   publishedAt         50 non-null     object 
 4   videoTitle          50 non-null     object 
 5   videoDescription    50 non-null     object 
 6   videoCategoryId     50 non-null     int64  
 7   videoCategoryLabel  50 non-null     object 
 8   duration            50 non-null     object 
 9   durationSec         50 non-null     int64  
 10  definition          50 non-null     object 
 11  caption             50 non-null     bool   
 12  viewCount           50 non-null     int64  
 13  likeCount           50 non-null     float64
 14  dislikeCount        50 non-null     float64
 15  commentCount        50 non-null     float64
dtypes: bool(1)

In [11]:
# checking for null values in the dataset..
# the "description" column has 6 null values
Ng_data.isna().sum()

title                       0
videoId                     0
channelId                   0
channelTitle                0
categoryId                  0
VideoCategoryDescription    0
Description                 6
viewCount                   0
likeCount                   0
dislikeCount                0
commentCount                0
favoriteCount               0
publishedAt                 0
duration                    0
dtype: int64

In [12]:
# checking for null values in the US data, the dataset has no null values.
US_data.isna().sum()

channelId             0
channelTitle          0
videoId               0
publishedAt           0
videoTitle            0
videoDescription      0
videoCategoryId       0
videoCategoryLabel    0
duration              0
durationSec           0
definition            0
caption               0
viewCount             0
likeCount             0
dislikeCount          0
commentCount          0
dtype: int64

In [13]:
# verifying that the null values in the description column has been removed
Ng_data = Ng_data.dropna()
print(Ng_data.isna().sum())

title                       0
videoId                     0
channelId                   0
channelTitle                0
categoryId                  0
VideoCategoryDescription    0
Description                 0
viewCount                   0
likeCount                   0
dislikeCount                0
commentCount                0
favoriteCount               0
publishedAt                 0
duration                    0
dtype: int64


In [14]:
# checking for null values after dropping the null values in the NG data
Ng_data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 44 entries, 0 to 49
Data columns (total 14 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   title                     44 non-null     object
 1   videoId                   44 non-null     object
 2   channelId                 44 non-null     object
 3   channelTitle              44 non-null     object
 4   categoryId                44 non-null     int64 
 5   VideoCategoryDescription  44 non-null     object
 6   Description               44 non-null     object
 7   viewCount                 44 non-null     int64 
 8   likeCount                 44 non-null     int64 
 9   dislikeCount              44 non-null     int64 
 10  commentCount              44 non-null     int64 
 11  favoriteCount             44 non-null     int64 
 12  publishedAt               44 non-null     object
 13  duration                  44 non-null     object
dtypes: int64(6), object(8)
memor

In [15]:
# # Subsetting columns to just the columns needed for the analysis of the NG data
Ng_data =Ng_data[["videoId", "VideoCategoryDescription", "Description", "viewCount" , "likeCount", "commentCount"]]

Ng_data.head()

videoId VideoCategoryDescription  \
0  Zv1i-Cdn4Ik                   Comedy   
1  BmCkK7W7j0Q         Film & Animation   
2  seZb9P7H6MU            Entertainment   
3  pbATe_uZFTE                    Music   
5  Ga_i97g0uFY                   Sports   

                                         Description  viewCount  likeCount  \
0  Please SUBSCRIBE to Broda Shaggi Official YouT...     654033      30422   
1  EP 13 "The Protest" - Olive leads a protest ag...     299169      21292   
2  Find out where the story leads in this latest ...     553886       5252   
3  Provided to YouTube by Starboy/RCA Records\n\n...     197492       8561   
5  The Nigerian striker smashes home the winner w...    1350957      21167   

   commentCount  
0          1474  
1          2012  
2           586  
3             0  
5          1043

In [16]:
# Subsetting columns to just the columns needed for the analysis of the US data

US_data =US_data[["videoId", 'videoDescription', "Description", "viewCount" , "likeCount", "commentCount"]]

US_data.head()

KeyError: "['VideoCategoryDescription', 'Description'] not in index"

In [ ]:
# describing the data

Ng_data.describe()

In [ ]:
# describing the data 

US_data.describe()

In [ ]:
# checking the correlation between the numeric variables

Ng_data.corr()

In [ ]:
# checking the correlation between the numeric variables

US_data.corr()

In [ ]:
# plotting viewCount against likecount
fig, ax=plt.subplots()
ax.scatter(Ng_data['likeCount'], Ng_data['viewCount'])
plt.xscale("log")
plt.yscale('log')
plt.style.use('seaborn-v0_8-colorblind')
plt.title("Visualizing the numeric variables")
plt.ylabel("viewCount", color= "Green")
plt.xlabel("likeCount", color= "g")
plt.show()

In [ ]:
# plotting ViewCount against commentCount

fig, ax=plt.subplots()
plt.scatter(Ng_data['commentCount'], Ng_data['viewCount'])
plt.xscale("log")
plt.yscale('log')
plt.style.use('seaborn-v0_8-colorblind')
plt.title("Visualizing the numeric Variables")
plt.ylabel("viewCount", color= "Green")
plt.xlabel("commentCount", color= "g")
plt.show()